bin/oneoff/script_clone_command_generator.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+# Generates the list of commands to clone each unit within an unit group
+# This assumes that the new unit group is already created
+
+require_relative '../../dashboard/config/environment'
+
+def generate_unit_clone_commands(source_unit_group_name, src_year, dstn_year)
+  destination_unit_group_name = source_unit_group_name.sub(src_year, dstn_year)
+
+  puts "\nUnit group: [#{source_unit_group_name}] for destination year [#{dstn_year}]"
+  puts "Create a destination unit group name with name [#{destination_unit_group_name}] at https://levelbuilder-studio.code.org/courses/new and run the commands below"
+
+  UnitGroup.find_by_name(source_unit_group_name).default_units.each do |unit_to_clone|
+    source_unit_name = unit_to_clone.name
+    destination_unit_name = source_unit_name.sub(src_year, dstn_year)
+    puts "Unit.find_by_name('#{source_unit_name}').clone_migrated_unit('#{destination_unit_name}', destination_unit_group_name: '#{destination_unit_group_name}', new_level_suffix: '#{dstn_year}')"
+  end
+end
+
+puts "\nThis script does not generate correct commands for deep learning courses, courses that belong to a course family and those which aren't versioned."
+puts "Pull latest from staging and run seeding to ensure the local curriculum state matches that of levelbuilder environment."
+print "\nEnter the source year:"
+source_year = gets.chomp
+
+print "Enter the destination year:"
+destination_year = gets.chomp
+
+print "Enter a comma seperated list of scripts to clone:"
+scripts_to_clone = gets.chomp
+
+scripts_to_clone.split(",").reject(&:blank?).map(&:strip).each do |scpt_to_clone|
+  clone_unit = Unit.find_by_name(scpt_to_clone)
+  unless clone_unit.nil?
+    dstn_unit_name = clone_unit.name.sub(source_year, destination_year)
+    family_name = clone_unit.family_name
+    puts "\nUnit: [#{scpt_to_clone}] for destination year [#{destination_year}]"
+    puts "Unit.find_by_name('#{scpt_to_clone}').clone_migrated_unit('#{dstn_unit_name}', family_name: '#{family_name}', version_year: '#{destination_year}', new_level_suffix: '#{destination_year}')"
```

Comment:
If we're hoping to automate these script clone jobs, would it make sense to display something more user friendly (i.e., `puts "Ready to clone #{scpt_to_clone} to #{dstn_unit_name}_#{destination_year}`) and have the script automatically run the generated commands after the user confirms?

Reply:
There are a few more steps like unit group creation and level builder content push that needs to be automated before we can run the cloning commands as part of this script.

I also think having some manual cross validation over the generated commands would be good give we have so many different cases before we fully automate the cloning.